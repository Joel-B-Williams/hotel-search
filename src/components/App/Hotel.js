import React, { Component } from 'react';

class Hotel extends Component {

  defaultImage = (ev) => {
    ev.target.src = "https://i.imgur.com/qUvaEAl.png"
  }

  render () {
    return (
      <div className="hotel-list">
        <img src={this.props.hotelStaticContent.mainImage.url} onError={this.defaultImage} className="photo"/>
        <div className="details">
          <div className="hotel-name">
            {this.props.hotelStaticContent.name || "No name found" }
          </div>
          <div className="hotel-location">
            {this.props.hotelStaticContent.neighborhoodName || "Location unknown"}
          </div>
        </div>
        <div className="price">
          <table className="price-table">
            <tbody>
              <tr>
                <td>Per night: </td>
                <td className="price-per-night">
                  {/* find less risky method if possible */}
                  <p dangerouslySetInnerHTML={{__html: this.props.lowestAveragePrice.symbol}}></p>
                  {this.props.lowestAveragePrice.amount || "$$"}
                </td>
              </tr>
              <tr className="points-row">
                <td>Points earned:</td>
                <td className="points-earned">
                  {this.props.rewards.miles || "Reward unavailable"}
                </td>
              </tr>
            </tbody>
          </table>
          <button className="btn hotel-btn">Select Now</button>
        </div>
      </div>
    );
  }
}
  
export default Hotel;