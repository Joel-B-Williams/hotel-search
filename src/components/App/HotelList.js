import React, { Component } from 'react';
import Hotel from './Hotel'

class HotelList extends Component {

  render() {
    const error = this.props.error;
    const errorMessage = 
    <div className="status-message">
      <p>Something has gone horribly pear-shaped</p>
      <p>Please try again</p>
    </div>
    
    return (
      <div>
        {error ? 
        (errorMessage) : 
        (this.props.hotels
        .filter(hotel => hotel.hotelStaticContent.name.toUpperCase().indexOf(this.props.searchTerm.toUpperCase()) >= 0)
        .map((hotel) => (<Hotel key={hotel.id} {...hotel} />)) )}
      </div>
    )        
  }
}


export default HotelList;