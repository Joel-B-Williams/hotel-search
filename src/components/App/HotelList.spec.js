import React from 'react';
import { shallow, mount } from 'enzyme';
import HotelList from './HotelList';
import { create } from 'react-test-renderer';
import Hotel from './Hotel';

describe('HotelList', () => {
  const list = [
    {"id":"907","rewards":{"miles":10000},"lowestAveragePrice":{"currency":"USD","symbol":"&#36;","amount":579.0},"hotelStaticContent":{"mainImage":{"category":"EXTERIOR","url":"http://d2whcypojkzby.cloudfront.net/imageRepo/2/0/68/56/314/ExteriorCarsGone_S.jpg"},"name":"Omni Chicago Hotel & Suites Magnificent Mile","neighborhoodName":"Magnificent Mile"}},
    {"id":"908","rewards":{"miles":8000},"lowestAveragePrice":{"currency":"USD","symbol":"&#36;","amount":234.0},"hotelStaticContent":{"mainImage":{"category":"EXTERIOR","url":"http://d2whcypojkzby.cloudfront.net/imageRepo/6/0/96/743/239/chidt-exterior-0227-hor-clsc_S.jpg"},"name":"Chicago Marriott Downtown Magnificent Mile","neighborhoodName":"Magnificent Mile"}},{"id":"87390","rewards":{"miles":12000},"lowestAveragePrice":{"currency":"USD","symbol":"&#36;","amount":123.0},"hotelStaticContent":{"mainImage":{"url":"http://d2whcypojkzby.cloudfront.net/imageRepo/1/0/16/99/327/012857_A_P.jpg"},"name":"Homewood Suites by Hilton Chicago-Lincolnshire","neighborhoodName":"Lincolnshire"}}
  ];

    const wrapper = shallow(<HotelList 
      hotels={list} 
      searchTerm={''}
      error={false} />);
    
    it('renders the component', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('matches snapshot', () => {
      const comp = create(<HotelList 
        hotels={list} 
        searchTerm={''}
        error={false} />);

      expect(comp.toJSON()).toMatchSnapshot();
    });

    it('shows all results when not filtered', () => {
      const wrapper = mount(<HotelList 
        hotels={list} 
        searchTerm={''}
        error={false} />);

      expect(wrapper.find('Hotel').length).toEqual(list.length);
      wrapper.unmount();
    });

    it('shows filtered results', () => {
      const wrapper = mount(<HotelList 
        hotels={list} 
        searchTerm={'magnificent'}
        error={false} />);

      expect(wrapper.find('Hotel').length).toEqual(2);
      wrapper.unmount();
    });
});