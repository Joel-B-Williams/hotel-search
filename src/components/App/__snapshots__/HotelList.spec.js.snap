// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList matches snapshot 1`] = `
<div>
  <div>
    <div
      className="hotel-list"
    >
      <img
        className="photo"
        onError={[Function]}
        src="http://d2whcypojkzby.cloudfront.net/imageRepo/2/0/68/56/314/ExteriorCarsGone_S.jpg"
      />
      <div
        className="details"
      >
        <div
          className="hotel-name"
        >
          Omni Chicago Hotel & Suites Magnificent Mile
        </div>
        <div
          className="hotel-location"
        >
          Magnificent Mile
        </div>
      </div>
      <div
        className="price"
      >
        <table
          className="price-table"
        >
          <tbody>
            <tr>
              <td>
                Per night: 
              </td>
              <td
                className="price-per-night"
              >
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#36;",
                    }
                  }
                />
                579
              </td>
            </tr>
            <tr
              className="points-row"
            >
              <td>
                Points earned:
              </td>
              <td
                className="points-earned"
              >
                10000
              </td>
            </tr>
          </tbody>
        </table>
        <button
          className="btn hotel-btn"
        >
          Select Now
        </button>
      </div>
    </div>
    <div
      className="hotel-list"
    >
      <img
        className="photo"
        onError={[Function]}
        src="http://d2whcypojkzby.cloudfront.net/imageRepo/6/0/96/743/239/chidt-exterior-0227-hor-clsc_S.jpg"
      />
      <div
        className="details"
      >
        <div
          className="hotel-name"
        >
          Chicago Marriott Downtown Magnificent Mile
        </div>
        <div
          className="hotel-location"
        >
          Magnificent Mile
        </div>
      </div>
      <div
        className="price"
      >
        <table
          className="price-table"
        >
          <tbody>
            <tr>
              <td>
                Per night: 
              </td>
              <td
                className="price-per-night"
              >
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#36;",
                    }
                  }
                />
                234
              </td>
            </tr>
            <tr
              className="points-row"
            >
              <td>
                Points earned:
              </td>
              <td
                className="points-earned"
              >
                8000
              </td>
            </tr>
          </tbody>
        </table>
        <button
          className="btn hotel-btn"
        >
          Select Now
        </button>
      </div>
    </div>
    <div
      className="hotel-list"
    >
      <img
        className="photo"
        onError={[Function]}
        src="http://d2whcypojkzby.cloudfront.net/imageRepo/1/0/16/99/327/012857_A_P.jpg"
      />
      <div
        className="details"
      >
        <div
          className="hotel-name"
        >
          Homewood Suites by Hilton Chicago-Lincolnshire
        </div>
        <div
          className="hotel-location"
        >
          Lincolnshire
        </div>
      </div>
      <div
        className="price"
      >
        <table
          className="price-table"
        >
          <tbody>
            <tr>
              <td>
                Per night: 
              </td>
              <td
                className="price-per-night"
              >
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#36;",
                    }
                  }
                />
                123
              </td>
            </tr>
            <tr
              className="points-row"
            >
              <td>
                Points earned:
              </td>
              <td
                className="points-earned"
              >
                12000
              </td>
            </tr>
          </tbody>
        </table>
        <button
          className="btn hotel-btn"
        >
          Select Now
        </button>
      </div>
    </div>
  </div>
  <div />
</div>
`;
